packages/@aws-cdk/aws-iam/lib/policy-statement.ts

```diff
@@ -473,14 +469,98 @@ export class PolicyStatement {
    */
   public validateForIdentityPolicy(): string[] {
     const errors = this.validateForAnyPolicy();
-    if (Object.keys(this.principal).length > 0 || Object.keys(this.notPrincipal).length > 0) {
+    if (Object.keys(this._principal).length > 0 || Object.keys(this._notPrincipal).length > 0) {
       errors.push('A PolicyStatement used in an identity-based policy cannot specify any IAM principals.');
     }
-    if (Object.keys(this.resource).length === 0 && Object.keys(this.notResource).length === 0) {
+    if (Object.keys(this._resource).length === 0 && Object.keys(this._notResource).length === 0) {
       errors.push('A PolicyStatement used in an identity-based policy must specify at least one resource.');
     }
     return errors;
   }
+
+  /**
+   * The Actions added to this statement
+   */
+  public get actions() {
+    return [...this._action];
+  }
+
+  /**
+   * The NotActions added to this statement
+   */
+  public get notActions() {
+    return [...this._notAction];
+  }
+
+  /**
+   * The Principals added to this statement
+   */
+  public get principals(): IPrincipal[] {
+    return [...this._principals];
+  }
+
+  /**
+   * The NotPrincipals added to this statement
+   */
+  public get notPrincipals(): IPrincipal[] {
+    return [...this._notPrincipals];
+  }
+
+  /**
+   * The Resources added to this statement
+   */
+  public get resources() {
+    return [...this._resource];
+  }
+
+  /**
+   * The NotResources added to this statement
+   */
+  public get notResources() {
+    return [...this._notResource];
+  }
+
+  /**
+   * The conditions added to this statement
+   */
+  public get conditions(): any {
+    return { ...this._condition };
+  }
+
+  /**
+   * Estimate the size of this policy statement
+   *
+   * By necessity, this will not be accurate. We'll do our best to overestimate
+   * so we won't have nasty surprises.
+   *
+   * @internal
+   */
+  public _estimateSize(): number {
+    let ret = 0;
+
+    const actionEstimate = 20;
+    const arnEstimate = 120; // A safe (over)estimate on how long ARNs typically are
```

Comment:
Per this doc: https://docs.aws.amazon.com/IAM/latest/APIReference/API_Policy.html

ARNs can be up to 2048 characters in length. The only way the estimate would be (dangerously) inaccurate is if the user imports an ARN, in which case they likely know the length of it themselves. Can we add an optional argument to ARN import methods, `arnLength?: number`, which we could use here? This way if users run into a deploy time issue they can specify the ARN length and `_estimateSize()` can correct for that here.